File: src/main/kotlin/com/presentation/intellij/tips/offer/infrastracture/api/OffersEndpoint.java
package com.presentation.intellij.tips.offer.infrastracture.api;

import com.presentation.intellij.tips.offer.Offer;
import com.presentation.intellij.tips.offer.Offers;
import com.presentation.intellij.tips.offer.OffersService;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.UUID;

@RestController
@RequestMapping
public class OffersEndpoint {

    private final OffersService offersService;

    public OffersEndpoint(OffersService offersService) {
        this.offersService = offersService;
    }

    @GetMapping("/offers")
    public Offers getOffers(
            @RequestParam("limit") Integer limit,
            @RequestParam("offset") Integer offset
    ) {
        List<Offer> offers = offersService.getOffers(limit, offset);
        return new Offers(offers);
    }

    @PostMapping("/users/{userId}/offers")
    public void addOffer(
            @PathVariable("userId") String userId,
            @RequestBody Offer offer
    ) {
        offersService.add(offer, userId, UUID.randomUUID().toString());
    }


}

File: src/main/kotlin/com/presentation/intellij/tips/offer/infrastracture/api/OffersEndpoint.kt
package com.presentation.intellij.tips.offer.infrastracture.api

import com.presentation.intellij.tips.offer.OffersService
import com.presentation.intellij.tips.offer.OffersServiceTwo
import org.springframework.web.bind.annotation.*
import java.net.URI

@RestController
@RequestMapping
class OffersEndpoint(
    private val offersService: OffersServiceTwo
) {

    @GetMapping("/offers")
    fun getOffers(
        @RequestParam("limit") limit: Int?,
        @RequestParam("offset") offset: Int?,
    ): Offers {
        val offers = offersService.getOffers(limit, offset)
        return Offers(offers)
    }

    @PostMapping("/users/{userId}/offers")
    fun addOffer(
        @PathVariable("userId") userId: String,
        @RequestBody offer: Offer
    ) {
        offersService.add(offer, userId)
    }
}

data class Offers(
    val offers: List<Offer>,
)

data class Offer(
    val id: Long,
    val title: String,
    val image: URI,
)
